docs/deployment/scaling.md

```diff
@@ -67,3 +67,131 @@ You must consider that the targets for CPU and Memory use the service replicas l
+
+
+Once the Datadog Cluster Agent is setup follow this [Datadog Documentation](https://docs.datadoghq.com/containers/guide/cluster_agent_autoscaling_metrics/?tab=daemonset#register-the-external-metrics-provider-service) to configure your Datadog Cluster Agent with an external metrics provider service.
+
+
+Next to setup DatadogMetric follow these steps from the [Datadog Documentation](https://docs.datadoghq.com/containers/guide/cluster_agent_autoscaling_metrics/?tab=daemonset#datadog-cluster-agent).
+
+You can now create scaling manifests based on Datadog's provided template:
+```html 
+apiVersion: datadoghq.com/v1alpha1
+kind: DatadogMetric
+metadata:
+  name: <your_datadogmetric_name>
+spec:
+  query: <your_custom_query>
+``` 
+You can deploy these metrics with `kubectl apply -f your-custom-metric-file.yaml`
+
+You will finally then need to deploy a service for your apps to push metrics to Datadog.  You can use this example or write your own based on your own custom configurations:
+```html 
+$ cat dd-agent-service.yaml
+apiVersion: v1 
+kind: Service 
+metadata: 
+  name: dd-agent 
+spec: 
+  selector: 
+    app: datadog 
+  ports: 
+    - protocol: UDP 
+      port: 8125 
+      targetPort: 8125 
+``` 
+Deploy the service with `kubectl apply -f dd-agent-service.yaml`
+
+
+In this example we will start with a [nodejs app](https://github.com/convox-examples/nodejs-dd-autoscale) that has been configured to scale with Datadog to demonstrate how to setup scaling with this method. 
+
+Based on the provided template we can make a basic page views metric for the node application: 
+```html 
+$ cat page-views-metrics.yaml
+apiVersion: datadoghq.com/v1alpha1 
+kind: DatadogMetric 
+metadata: 
+  name: page-views-metrics 
+spec: 
+  query: avg:page.views{*}.as_count() 
+```
+Then add the metric to your cluster via `kubectl apply -f page-views-metrics.yaml`
+
+
+Now we are going to look at the configuration of the `convox.yaml` from our nodejs app to autoscale based on this newly created metric.
+
+ We will specify the `datadogmetric` object name in the `scale.targets[].external section`. The `datadogmetric` object name convention to use in `scale.targets[].external` section is : `datadogmetric@<namespace>:<datadogmetric_name>`
+
+We set the env var `DD_AGENT_HOST` to point at the dd-agent service we previously made to push information to Datadog.
+
+```html 
+environment:
+  - PORT=3000
+services:
+  web:
+    build: .
+    port: 3000
+    environment:
+      - DD_AGENT_HOST=dd-agent.default.svc.cluster.local
+    scale:
+      count: 1-3
+      targets:
+        external:
+          - name: "datadogmetric@default:page-views-metrics"
+            averageValue: 5
+``` 
+[Configured Nodejs app](https://github.com/convox-examples/nodejs-dd-autoscale)
+
+You can now deploy the application with `convox apps create` then `convox deploy`
```

Comment:
`convox apps create -a nodejs` then `convox deploy -a nodejs`
since by default takes the directory name and this has `https://github.com/convox-examples/nodejs-dd-autoscale` different name.
or you can change the `-a nodejs` on the below commands